File: r/experiments/expr_get_unique_factor_level.R
# Created on Fri Oct  8 11:02:08 2021
# Objective: Given a dataframe, Get a list of categories for categorical variable
# @author: Ashish

# create some fake data
df <- data.frame(Name = c("Jon", "Bill", "Maria", "Ben", "Tina"),
                 Age = c(23, 41, 32, 58, 26),
                 sex = c('M','T','F','M','F'))
str(df)
# coerce column to categorical
df$sex<- as.factor(df$sex)
str(df)

# solution 1: using sapply()
sapply(df, levels)

# solution 2: using pipe operator
library(dplyr)
df %>%
  sapply(levels)

File: python_3/experiments/expr_get_unique_factor_level.py
# -*- coding: utf-8 -*-
"""
Created on Fri Oct  8 11:02:08 2021
Objective: Given a dataframe, Get a list of categories for categorical variable
@author: Ashish
"""

# load required libraries
import pandas as pd
import numpy as np

# create some fake data
a = ['arrived','departed','delayed']
df = pd.DataFrame(np.random.choice(a, size=(10,3)), 
                  columns = ['Col1','Col2','Col3'])
print(df)
print(df.dtypes)

# coerce object data type to categorical format
df['Col1'] = pd.Categorical(df['Col1'])
print(df.dtypes)

# show unique categoris
print(df['Col1'].unique())



